File: api/src/test/java/net/blueshell/api/factory/model/ContributionPeriodFactory.java
package net.blueshell.api.factory.model;

import com.github.javafaker.Faker;
import lombok.RequiredArgsConstructor;
import net.blueshell.api.model.contribution.ContributionPeriod;
import org.springframework.stereotype.Component;

import java.time.LocalDate;
import java.util.concurrent.atomic.AtomicLong;

/**
 * Factory for ContributionPeriod model test instances.
 */
@Component
@RequiredArgsConstructor
public class ContributionPeriodFactory {

    private static final AtomicLong counter = new AtomicLong(1000);
    private final Faker faker;

    public ContributionPeriod createBasic() {
        ContributionPeriod cp = new ContributionPeriod();
        cp.setId(generateId());
        LocalDate start = LocalDate.now().withDayOfMonth(1);
        cp.setStartDate(start);
        cp.setEndDate(start.plusMonths(6));
        cp.setHalfYearFee(10.0);
        cp.setFullYearFee(18.0);
        cp.setAlumniFee(5.0);
        cp.setListId(faker.number().randomNumber());
        return cp;
    }

    public ContributionPeriod createFull() {
        return createBasic();
    }

    public ContributionPeriod createWithCustomizations(java.util.function.Consumer<ContributionPeriod> customizer) {
        ContributionPeriod cp = createFull();
        customizer.accept(cp);
        return cp;
    }

    private Long generateId() {
        return counter.incrementAndGet();
    }
}

File: api/src/test/kotlin/net/blueshell/api/factory/model/contribution/ContributionPeriodFactory.kt
package net.blueshell.api.factory.model.contribution

import com.github.javafaker.Faker
import net.blueshell.api.domain.contribution.persistence.ContributionPeriod
import org.springframework.stereotype.Component
import java.time.LocalDate
import java.util.concurrent.atomic.AtomicLong
import java.util.function.Consumer

/**
 * Factory for ContributionPeriod model test instances.
 */
@Component
class ContributionPeriodFactory(
    private val faker: Faker
) {

    fun createBasic(): net.blueshell.api.domain.contribution.persistence.ContributionPeriod {
        val period = _root_ide_package_.net.blueshell.api.domain.contribution.persistence.ContributionPeriod()
        val start = LocalDate.now().withDayOfMonth(1)
        period.startDate = start
        period.endDate = start.plusMonths(6)
        period.halfYearFee = 10.0
        period.fullYearFee = 18.0
        period.alumniFee = 5.0
        period.listId = faker.number().randomNumber()
        return period
    }

    fun createFull(): net.blueshell.api.domain.contribution.persistence.ContributionPeriod = createBasic()

    fun createWithCustomizations(customizer: Consumer<net.blueshell.api.domain.contribution.persistence.ContributionPeriod>): net.blueshell.api.domain.contribution.persistence.ContributionPeriod {
        val period = createFull()
        customizer.accept(period)
        return period
    }

    private fun generateId(): Long = COUNTER.incrementAndGet()

    private companion object {
        val COUNTER = AtomicLong(1000)
    }
}
